Gems/LandscapeCanvas/Code/Source/Editor/MainWindow.cpp

```diff
@@ -1558,6 +1572,33 @@ namespace LandscapeCanvasEditor
         });
     }
 
+    void MainWindow::HandleSetImageAssetPath(const AZ::EntityId& sourceEntityId, const AZ::EntityId& targetEntityId)
+    {
+        // This only gets called when a valid connection is made between a Gradient Baker output image slot
+        // (sourceEntityId) and an Image Gradient input image asset slot (targetEntityId)
+        // So we need to use the corresponding request bus APIs to update the image asset path on
+        // the Image Gradient
+        AZ::IO::Path outputImagePath;
+        GradientSignal::GradientBakerRequestBus::EventResult(
+            outputImagePath, sourceEntityId, &GradientSignal::GradientBakerRequests::GetOutputImagePath);
+
+        if (!outputImagePath.empty())
+        {
+            AzToolsFramework::ScopedUndoBatch undo("Update Image Gradient Asset");
+
+            // The ImageGradientRequests::SetImageAssetPath only takes a product path, but we are given
+            // a source asset path, so need to append the product extension
+            QString imageAssetPath = QString::fromUtf8(
+                outputImagePath.c_str(), static_cast<int>(outputImagePath.Native().size()));
+            imageAssetPath += ".streamingimage";
```

Comment:
This is happening in the gradient baker too, and possibly in some python scripts - should we have a "SetImageAssetSourcePath" or something that will do this work in one place?

Reply:
Yes this is a good idea, I can do that in this PR since I need to update it anyway with other feedback. I can update the python tests so that they exercise both the source and product path APIs.

Reply:
Ended up leaving the python tests as-is because they are also hitting the asset catalog APIs directly which only deal with the product extension as well. But I updated the usage in the gradient baker and in Landscape Canvas to use the new source asset path APIs so they are getting exercised.